File: src/AndroidNativeBenchmark/app/src/main/java/com/rochapires/admin/androidnativebenchmark/Activities/ActivityAlgorithmPerformance.java
package com.rochapires.admin.androidnativebenchmark.Activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import com.rochapires.admin.androidnativebenchmark.R;


public class ActivityAlgorithmPerformance extends AppCompatActivity {

    private TextView textViewlTime1;
    private TextView textViewlTime2;
    private TextView textViewlTime3;
    private TextView textViewlTime4;
    private TextView textViewlTime5;
    private TextView textViewlTime6;
    private TextView textViewlTime7;
    private TextView textViewlTime8;
    private TextView textViewlTime9;
    private TextView textViewlTime10;
    private TextView textViewlTimeAverage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_activity_algorithm_performance);

        textViewlTime1 = (TextView) findViewById(R.id.textViewlTime1);
        textViewlTime2 = (TextView) findViewById(R.id.textViewlTime2);
        textViewlTime3 = (TextView) findViewById(R.id.textViewlTime3);
        textViewlTime4 = (TextView) findViewById(R.id.textViewlTime4);
        textViewlTime5 = (TextView) findViewById(R.id.textViewlTime5);
        textViewlTime6 = (TextView) findViewById(R.id.textViewlTime6);
        textViewlTime7 = (TextView) findViewById(R.id.textViewlTime7);
        textViewlTime8 = (TextView) findViewById(R.id.textViewlTime8);
        textViewlTime9 = (TextView) findViewById(R.id.textViewlTime9);
        textViewlTime10 = (TextView) findViewById(R.id.textViewlTime10);
        textViewlTimeAverage = (TextView) findViewById(R.id.textViewlTimeAverage);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_activity_algorithm_performance, menu);
        return true;
    }

    public void RunFibonacci(View view)
    {
        runOnUiThread(new Runnable() {
            public void run() {
                TextView textViewlTime = (TextView) findViewById(R.id.textViewlTime1);
                EditText editTextNumberOfIterations = (EditText) findViewById(R.id.editTextNumberOfIterations);
                int numberOfIterations = Integer.parseInt(editTextNumberOfIterations.getText().toString());

                long startTime = System.currentTimeMillis();
                int result = Fibonacci(numberOfIterations);
                long timeElapsed = System.currentTimeMillis() - startTime;

                textViewlTime1.setText("Time " + ": " + timeElapsed);
                textViewlTime2.setText("");
                textViewlTime3.setText("");
                textViewlTime4.setText("");
                textViewlTime5.setText("");
                textViewlTime6.setText("");
                textViewlTime7.setText("");
                textViewlTime8.setText("");
                textViewlTime9.setText("");
                textViewlTime10.setText("");
                textViewlTimeAverage.setText("");
            }
        });
    }

    public void RunFibonacciTenTimes(View view)
    {
        runOnUiThread(new Runnable() {
            public void run() {

                EditText editTextNumberOfIterations = (EditText) findViewById(R.id.editTextNumberOfIterations);
                int numberOfIterations = Integer.parseInt(editTextNumberOfIterations.getText().toString());

                long totalTime = 0;
                for(int i = 1; i <= 10; i++) {
                    long startTime = System.currentTimeMillis();
                    int result = Fibonacci(numberOfIterations);
                    long timeElapsed = System.currentTimeMillis() - startTime;
                    totalTime += timeElapsed;
                    String timeElapsedText = "Time " + i + ": " + timeElapsed;
                    switch (i){
                        case 1:
                            textViewlTime1.setText(timeElapsedText);
                            break;
                        case 2:
                            textViewlTime2.setText(timeElapsedText);
                            break;
                        case 3:
                            textViewlTime3.setText(timeElapsedText);
                            break;
                        case 4:
                            textViewlTime4.setText(timeElapsedText);
                            break;
                        case 5:
                            textViewlTime5.setText(timeElapsedText);
                            break;
                        case 6:
                            textViewlTime6.setText(timeElapsedText);
                            break;
                        case 7:
                            textViewlTime7.setText(timeElapsedText);
                            break;
                        case 8:
                            textViewlTime8.setText(timeElapsedText);
                            break;
                        case 9:
                            textViewlTime9.setText(timeElapsedText);
                            break;
                        case 10:
                            textViewlTime10.setText(timeElapsedText);
                            break;
                        default:
                            break;
                    }
                }
                textViewlTimeAverage.setText("Average Time: " + totalTime / 10);
                Log.Info("AlgorithmPerformance", textViewlTimeAverage.Text);
            }
        });
    }

    public static int Fibonacci(int number)
    {
        if (number == 0) {
            return 0;
        } else if (number == 1) {
            return 1;
        } else
        {
            return Fibonacci(number - 2) + Fibonacci(number - 1);
        }
    }

}

File: src/AndroidXamarinBenchmark/AndroidXamarinBenchmark/Activities/ActivityAlgorithmPerformance.cs
using Android.App;
using Android.OS;
using Android.Util;
using Android.Widget;
using System;
using System.Diagnostics;

namespace AndroidXamarinBenchmark.Activities
{

    [Activity(Label = "ActivityAlgorithmPerformance")]
    public class ActivityAlgorithmPerformance : Activity
    {
		private TextView textViewlTime1;
		private TextView textViewlTime2;
		private TextView textViewlTime3;
		private TextView textViewlTime4;
		private TextView textViewlTime5;
		private TextView textViewlTime6;
		private TextView textViewlTime7;
		private TextView textViewlTime8;
		private TextView textViewlTime9;
		private TextView textViewlTime10;
		private TextView textViewlTimeAverage;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            SetContentView(Resource.Layout.ActivityAlgorithmPerformance);

			Button buttonRun = FindViewById<Button>(Resource.Id.buttonRun);
			Button buttonRunTenTimes = FindViewById<Button>(Resource.Id.buttonRunTenTimes);

			EditText editTextNumberOfIterations = FindViewById<EditText>(Resource.Id.editTextNumberOfIterations);

			textViewlTime1 = FindViewById<TextView>(Resource.Id.textViewlTime1);
			textViewlTime2 = FindViewById<TextView>(Resource.Id.textViewlTime2);
			textViewlTime3 = FindViewById<TextView>(Resource.Id.textViewlTime3);
			textViewlTime4 = FindViewById<TextView>(Resource.Id.textViewlTime4);
			textViewlTime5 = FindViewById<TextView>(Resource.Id.textViewlTime5);
			textViewlTime6 = FindViewById<TextView>(Resource.Id.textViewlTime6);
			textViewlTime7 = FindViewById<TextView>(Resource.Id.textViewlTime7);
			textViewlTime8 = FindViewById<TextView>(Resource.Id.textViewlTime8);
			textViewlTime9 = FindViewById<TextView>(Resource.Id.textViewlTime9);
			textViewlTime10 = FindViewById<TextView>(Resource.Id.textViewlTime10);
			textViewlTimeAverage = FindViewById<TextView>(Resource.Id.textViewlTimeAverage);

			buttonRun.Click += (sender, e) =>
			{
				RunOnUiThread(() =>
					{
						Stopwatch sw = new Stopwatch();
						int numberOfIterations = int.Parse(editTextNumberOfIterations.Text);

						//just to warm things up
						Fibonacci(10);

						sw.Start();
						int result = Fibonacci(numberOfIterations);
						sw.Stop();

						textViewlTime1.Text = "Time: " + sw.ElapsedMilliseconds;
						textViewlTime2.Text = "";
						textViewlTime3.Text = "";
						textViewlTime4.Text = "";
						textViewlTime5.Text = "";
						textViewlTime6.Text = "";
						textViewlTime7.Text = "";
						textViewlTime8.Text = "";
						textViewlTime9.Text = "";
						textViewlTime10.Text = "";
						textViewlTimeAverage.Text = "";
					});
			};

			buttonRunTenTimes.Click += (sender, e) =>
			{
				RunOnUiThread(() =>
					{
						Stopwatch sw = new Stopwatch();
						int numberOfIterations = int.Parse(editTextNumberOfIterations.Text);

						//just to warm things up
						Fibonacci(10);

						long totalTime = 0;
						for(int i = 1; i <= 10; i++) {
							sw.Start();
							int result = Fibonacci(numberOfIterations);
							sw.Stop();

							long elapsedMilli = sw.ElapsedMilliseconds;
							totalTime += elapsedMilli;
							string timeElapsedText = "Time " + i + ": " + elapsedMilli;
							switch(i) {
							case 1:
								textViewlTime1.Text = timeElapsedText;
								break;
							case 2:
								textViewlTime2.Text = timeElapsedText;
								break;
							case 3:
								textViewlTime3.Text = timeElapsedText;
								break;
							case 4:
								textViewlTime4.Text = timeElapsedText;
								break;
							case 5:
								textViewlTime5.Text = timeElapsedText;
								break;
							case 6:
								textViewlTime6.Text = timeElapsedText;
								break;
							case 7:
								textViewlTime7.Text = timeElapsedText;
								break;
							case 8:
								textViewlTime8.Text = timeElapsedText;
								break;
							case 9:
								textViewlTime9.Text = timeElapsedText;
								break;
							case 10:
								textViewlTime10.Text = timeElapsedText;
								break;
							default:
								break;
                            }
                            Log.Info("AlgorithmPerformance", timeElapsedText);

                            sw.Reset();
						}
						textViewlTimeAverage.Text = "Average Time: " + totalTime / 10;
                        Log.Info("AlgorithmPerformance", textViewlTimeAverage.Text);
                    });
			};
        }

        public static int Fibonacci(int number)
        {
            if (number == 0) {
                return 0;
            } else if (number == 1) {
                return 1;
            } else
            {
                return Fibonacci(number - 2) + Fibonacci(number - 1);
            }
        }

    }

}